File: functions/[[path]].py
from flask import Flask, Request
from app import app as flask_app

def onRequest(context):
    """
    Entry point for Cloudflare Pages Functions
    """
    try:
        request = context.get('request', {})
        # Create Flask request
        flask_request = Request.from_values(
            base_url=request.get('url', ''),
            method=request.get('method', 'GET'),
            headers=dict(request.get('headers', {})),
            input_stream=request.get('body', '').encode() if request.get('body') else None
        )
        
        # Process the request through Flask
        with flask_app.request_context(flask_request):
            response = flask_app.full_dispatch_request()
            return {
                'statusCode': response.status_code,
                'headers': dict(response.headers),
                'body': response.get_data(as_text=True)
            }
    except Exception as e:
        return {
            'statusCode': 500,
            'body': f'Internal Server Error: {str(e)}',
            'headers': {'Content-Type': 'text/plain'}
        } 

File: functions/[[path]].js
export async function onRequest(context) {
  const { request, env } = context;
  
  try {
    // First try to serve static assets
    const url = new URL(request.url);
    const path = url.pathname;
    
    // Try to serve from static assets first
    try {
      const asset = await env.ASSETS.fetch(request);
      if (asset.status !== 404) {
        return asset;
      }
    } catch (e) {
      // Continue to dynamic handling if static asset not found
    }

    // For dynamic routes, return the index.html
    if (!path.startsWith('/static/') && !path.endsWith('.ico')) {
      const response = await env.ASSETS.fetch(`${url.origin}/index.html`);
      return new Response(response.body, {
        headers: {
          'content-type': 'text/html;charset=UTF-8',
        },
      });
    }

    // If nothing matched, return 404
    return new Response('Not Found', { status: 404 });
  } catch (err) {
    return new Response(`Internal Server Error: ${err.message}`, { 
      status: 500,
      headers: { 'content-type': 'text/plain' }
    });
  }
} 